File: TriviaGame.Web/src/GameSessionInfo.ts
import {TriviaAnswer, TriviaBoard} from '@/models.g';
import {Player, Team} from '@/Player';

export class GameSessionInfo {

  public get totalPointsTeamOne(): number {
    return this.triviaBoard.answers
      ?.filter((value: TriviaAnswer) => value.wonBy === Team.One)
      ?.map((value: TriviaAnswer) => value.points)
      ?.reduce((sum, current) => (sum ?? 0) + (current ?? 0), 0) ?? 0;
  }

  public get totalPointsTeamTwo(): number {
    return this.triviaBoard.answers
      ?.filter((value: TriviaAnswer) => value.wonBy === Team.Two)
      ?.map((value: TriviaAnswer) => value.points)
      ?.reduce((sum, current) => (sum ?? 0) + (current ?? 0), 0) ?? 0;
  }

  public get teamOnePlayers(): Player[] {
    return this.players.filter((value: Player) => value.team === Team.One);
  }

  public get teamTwoPlayers(): Player[] {
    return this.players.filter((value: Player) => value.team === Team.Two);
  }

  public gameId: string | null;

  public players: Player[];

  public triviaBoard: TriviaBoard;

  public totalAnswers: number;

  public host: Player | null;
  constructor() {
    this.gameId = null;
    this.players = [];
    this.triviaBoard = new TriviaBoard();
    this.totalAnswers = 0;
    this.host = null;
  }
}

File: TriviaGame.Data/Services/Impl/GameSessionInfo.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using TriviaGame.Data.Models;

namespace TriviaGame.Data.Services.Impl
{
    public class GameSessionInfo
    {
        public GameSessionInfo(string gameId)
        {
            GameId = gameId;
        }

        public GameSessionInfo(string gameId, params Player[] players) : this(gameId)
        {
            foreach (var player in players)
            {
                PlayerIds.Add(player.ConnectionId, player);
            }
        }

        public string GameId { get; set; }

        [JsonIgnore]
        public Dictionary<string, Player> PlayerIds { get; set; } = new();

        public IEnumerable<Player> Players => PlayerIds.Select(x => x.Value);

        public TriviaBoard TriviaBoard { get; set; }

        public int TotalAnswers { get; set; }

        public Player Host { get; set; }
    }

    public class Player
    {
        public string ConnectionId { get; set; }

        public Team Team { get; set; }
    }

    public enum Team
    {
        Unknown = 0,
        One = 1,
        Two = 2
    }
}
